src/categories/components/CategoryProductList/CategoryProductList.tsx

```diff
@@ -89,58 +90,29 @@ export const CategoryProductList: React.FC<CategoryProductListProps> = props =>
     toolbar,
     onNextPage,
     onPreviousPage,
-    onRowClick
+    onRowClick,
+    selectedChannel
   } = props;
 
   const classes = useStyles(props);
-  const intl = useIntl();
 
   const numberOfColumns = 5;
 
-  // const getProductPrice = (
-  //   priceRangeUndiscounted: CategoryDetails_category_products_edges_node_pricing_priceRangeUndiscounted
-  // ) => {
-  //   if (!priceRangeUndiscounted) {
-  //     return null;
-  //   }
-
-  //   const { start, stop } = priceRangeUndiscounted;
-  //   const {
-  //     gross: { amount: startAmount }
-  //   } = start;
-  //   const {
-  //     gross: { amount: stopAmount }
-  //   } = stop;
-
-  //   if (startAmount === stopAmount) {
-  //     return (
-  //       <Money
-  //         money={{
-  //           amount: startAmount,
-  //           currency: start.gross.currency
-  //         }}
-  //       />
-  //     );
-  //   } else {
-  //     return (
-  //       <>
-  //         <Money
-  //           money={{
-  //             amount: startAmount,
-  //             currency: start.gross.currency
-  //           }}
-  //         />
-  //         {" - "}
-  //         <Money
-  //           money={{
-  //             amount: stopAmount,
-  //             currency: stop.gross.currency
-  //           }}
-  //         />
-  //       </>
-  //     );
-  //   }
-  // };
+  const getProductPrice = (channel: ChannelListingProductFragment) => {
+    if (channel?.discountedPrice) {
+      return (
+        <Money
+          money={{
+            amount: channel.discountedPrice?.amount,
+            currency: channel.discountedPrice?.currency
+              ? channel.discountedPrice?.currency
+              : channel.channel.currencyCode
```

Comment:
Could be shorter `channel.discountedPrice?.currency || channel.channel.currencyCode`